src/transformers/generation_tf_logits_process.py

```diff
@@ -259,8 +255,8 @@ def _create_score_penalties(self, input_ids, logits):
             np.put(token_penalties[i], prev_input_id, logit_penalties)
         return tf.convert_to_tensor(token_penalties, dtype=tf.float32)
 
-    def __call__(self, input_ids: tf.Tensor, scores: tf.Tensor) -> tf.Tensor:
-        score_penalties = self._create_score_penalties(input_ids, scores)
+    def __call__(self, input_ids: tf.Tensor, scores: tf.Tensor, cur_len: int) -> tf.Tensor:
```

Comment:
XLA greedy search was probably missing this as well in the logits processors, since it has the same padded `input_ids`

Reply:
Yes thanks for adding it! Just note that I don't really think we can make this processor XLA compilable anyways as it's very complex and numpy can't be used in XLA. `cur_len` is mostly added in Flax/JAX to make the rprocessors XLA-compilable. But doesn't hurt to added it here!

Reply:
`tf.unique` is not compatible with XLA because the output shape is dependent on the specific input data, and so cannot be inferred at compile time. However, there should be a way to make this logit processor XLA-compilable - there's probably some solution where you store counts in a sparse matrix and then use `triu()` or `tril()` followed by a matmul to see if a logit has been preceded by that same logit. Let me know if you want me to try that (here or in a separate PR)

Reply:
🧠 
I'd leave it to a subsequent PR, XLA-readiness is not the main priority here and this PR is already very long